github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitMessageCheck.java

```diff
@@ -0,0 +1,130 @@
+package com.github.kostyasha.github.integration.branch.events.impl;
+
+import com.github.kostyasha.github.integration.branch.GitHubBranchCause;
+import com.github.kostyasha.github.integration.branch.GitHubBranchRepository;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchCommitCheck;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchCommitCheckDescriptor;
+import hudson.Extension;
+import org.kohsuke.github.GHBranch;
+import org.kohsuke.github.GHCompare.Commit;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.CheckForNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GitHubBranchCommitMessageCheck extends GitHubBranchCommitCheck {
+    private static final String DISPLAY_NAME = "Commit Message Pattern";
+
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubBranchCommitMessageCheck.class);
+
+    private boolean exclude;
+
+    private Set<String> matchCriteria;
+
+    @DataBoundConstructor
+    public GitHubBranchCommitMessageCheck() {
+        this.matchCriteria = Collections.emptySet();
+    }
+
+    @Override
+    public GitHubBranchCause check(GHBranch remoteBranch, GitHubBranchRepository localRepo, Commit[] commits) {
+        String name = remoteBranch.getName();
+        if (matchCriteria.isEmpty()) {
+            LOG.warn("Commit message event added but no match criteria set, all commits are allowed.");
+            return null;
+        }
+
+        List<String> messages = Stream.of(commits)
+                .map(commit -> commit.getCommit().getMessage())
+                .collect(Collectors.toList());
+
+        if (commitsAreAllowed(messages)) {
+            LOG.debug("Commit messages {} for branch [{}] allowed, commit ignored.", messages, name);
+            return null;
+        }
+
+        return toCause(remoteBranch, localRepo, true, "Commit messages %s for branch [%s] not allowed by check.", messages, name);
```

Comment:
i updated the javadoc to explain this. i guess this makes this 'event/check' really more like a filter, so it would need to come higher in the list of definitions.